core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -540,11 +544,26 @@ private void updateCacheStream(long blockId) throws IOException {
     }
 
     try {
-      WorkerNetAddress address = mLocationPolicy.getWorkerForNextBlock(
-          mBlockStore.getWorkerInfoList(), getBlockSizeAllocation(mPos));
-      // If we reach here, we need to cache.
-      mCurrentCacheStream =
-          mBlockStore.getOutStream(blockId, getBlockSize(mPos), address, mOutStreamOptions);
+      List<BlockWorkerInfo> workers = mBlockStore.getWorkerInfoList();
```

Comment:
1. why do we still need to cache when the block is cached remotely? 
2. If we want to cache, i think it is better to avoid this getWorkerInfoList RPC which is per block. We can just pass the location to cache to getOutStream which has the worker list information.